src/Compilers/CSharp/Test/Emit/Emit/EndToEndTests.cs

```diff
@@ -316,5 +316,70 @@ static void runTest(int n)
                 });
             }
         }
+
+        [WorkItem(51739, "https://github.com/dotnet/roslyn/issues/51739")]
+        [ConditionalFact(typeof(WindowsOrLinuxOnly))]
+        public void NullableAnalysisNestedExpressionsInMethod()
+        {
+            NullableAnalysisNestedExpressions(insideLocalFunction: false);
+        }
+
+        [WorkItem(51739, "https://github.com/dotnet/roslyn/issues/51739")]
+        [ConditionalFact(typeof(WindowsOrLinuxOnly))]
+        public void NullableAnalysisNestedExpressionsInLocalFunction()
+        {
+            NullableAnalysisNestedExpressions(insideLocalFunction: true);
+        }
+
+        private static void NullableAnalysisNestedExpressions(bool insideLocalFunction)
```

Comment:
This test doesn't directly hit the failure case. (A version of this test that did hit the failure case was too specific to be usable as a general test for different platforms etc.) I'll see if there are other alternatives.

Reply:
Have you been able to verify the fix via ad-hoc testing?

Reply:
> Have you been able to verify the fix via ad-hoc testing?

Yes. I verified the failure and the fix with a version of this test locally. But the test was not sufficiently general to hit the failure on all platforms, etc.

---
In reply to: [593512691](https://github.com/dotnet/roslyn/pull/51851#discussion_r593512691) [](ancestors = 593512691)

Reply:
The latest tests test the issue directly.

---
In reply to: [593513744](https://github.com/dotnet/roslyn/pull/51851#discussion_r593513744) [](ancestors = 593513744,593512691)